src/models/instance.ts

```diff
@@ -34,6 +35,17 @@ export default class Instance {
         return new Instance(initialData);
     }
 
+    public static getOrCreate(data: Data, encryptionKey: string): Instance {
+        let instance;
+        if (!!data) {
```

Comment:
While this is conceptually perfect, `if (boolean)`, in dynamic languages is ok to leave the object: `if (object)`, because we simply are checking if it has a "truthy" value (according to the language semantics). It's also visually less distracting.